### PR TITLE
Update to latest OS packages and add support for Ubuntu, Fedora, and AL 2023

### PR DESCRIPTION
**Pull request by jsf9k**
## 🗣 Description ##

This pull request updates the Ansible role to:
- Use the latest versions of the OS packages
- Support Ubuntu, Fedora, and Amazon Linux 2023

## 💭 Motivation and context ##

- We may as well use the latest versions of the OS packages as opposed to relying on CDM to update them for us.  (Also our packages were woefully outdated - two major versions behind!)
- Supporting Fedora will help us to add CrowdStrike support to [our FreeIPA server AMI](https://github.com/cisagov/freeipa-server-packer).

## 🧪 Testing ##

All automated testing passes.

## ✅ Pre-approval checklist ##

- [x] This PR has an informative and human-readable title.
- [x] Changes are limited to a single goal - *eschew scope creep!*
- [x] All relevant type-of-change labels have been added.
- [x] I have read the [CONTRIBUTING](../blob/develop/CONTRIBUTING.md) document.
- [x] These code changes follow [cisagov code standards](https://github.com/cisagov/development-guide).
- [x] All new and existing tests pass.

### CODE REVIEW

File: tasks/RedHat.yml [hunk 1]
```diff
@@ -0,0 +1,36 @@
+---
+- name: Grab CrowdStrike Falcon system package GPG key from S3
+  ansible.builtin.aws_s3:
+    bucket: "{{ crowdstrike_third_party_bucket_name }}"
+    object: "{{ package_gpg_key_object_name }}"
+    dest: /tmp/{{ package_gpg_key_object_name }}
+    mode: get
```
Comment (dav3r) [inline tasks/RedHat.yml]: Alphabetize keys:
```suggestion
    bucket: "{{ crowdstrike_third_party_bucket_name }}"
    dest: /tmp/{{ package_gpg_key_object_name }}
    mode: get
    object: "{{ package_gpg_key_object_name }}"
```

**Review by dav3r — APPROVED** [review]
Looks great - only one small quibble. 👍

File: vars/Amazon.yml [hunk 2]
```diff
@@ -0,0 +1,8 @@
+---
+# The name of the S3 object corresponding to the CrowdStrike Falcon
+# sensor system package
+package_object_name: falcon-sensor-6.56.0-15309.amzn2023.x86_64.rpm
+
+# The name of the S3 object corresponding to the GPG key for the
+# CrowdStrike Falcon system package
+package_gpg_key_object_name: "Falcon Linux Sensor code signing certificate 2021.pem"
```
Comment (dav3r) [inline vars/Amazon.yml]: Can this be deleted since we are not using it?
```suggestion
```

Comment (dav3r) [reply vars/Amazon.yml]: Alternatively, we could comment it out and add a TODO linked to an issue.

**Review by dav3r — COMMENTED** [review]
I have one note before I approve this.

**Review by dav3r — APPROVED** [review]
🚀

**Review by dv4harr10 — APPROVED** [review]

**Pull request merged**